### ISSUE TITLE
Update player contributions for next season

### ISSUE THREAD

**Issue by minj** [minj/foxtrick#1194]
Original [issue 1194](https://code.google.com/p/foxtrick/issues/detail?id=1194) created by [Grebliux](mailto:Grebliux@gmail.com) on 2014-04-12T23:59:07.000Z:

I implemented a feature which calculates how much a player contributes to each position and finds the best one for him. All calculations are made using: http://wiki.hattrick.org/wiki/Hattrick_-_Skill_positions

Currently contributions to all positions are shown as a table in playerDetails page at the bottom.

In ownPlayers and transferSearchResult pages best position is shown for each player. Just one line of text under HTMS points. 

I also added new columns of each and best positions into skill table. So you can check updated skill table in ownPlayers and transferSearchResult pages. 

Here I attach archive of all files I changed/updated or created. I do not included files which I didn't touch in foxtrick. 

List of all changes: 
    Created module: /information-aggregation/best-player-position.js
    Created module: /information-aggregation/player-positions-evaluations.js
    Added method Foxtrick.Pages.Player.getSpeciality(doc)
    Added method Foxtrick.Pages.Player.getPositionsContributions(skills, spec)
    Added method Foxtrick.Pages.Player.getBestPosition(contributions)
    Created method addContributionsInfo() in Foxtrick.Pages.Players
    Slightly modified transfer-search-results.js
    Modified skill-table.js module
    Added new default preferences in default/preferences/foxtrick.js
    Added new language strings into content/foxtrick.properties

Could someone have a look, please? I would be more than happy if you would include this into next Foxtrick release.

**Comment by minj** [comment]
Should be done for next season. I'll create a new issue for match order

I keep asking more feedback but nothing valuable so far:

**From:** LA-MJ
**PostID:** [16741488.522](https://www.hattrick.org/goto.ashx?path=%2FForum%2FRead.aspx%3Ft%3D16741488%26n%3D522%26v%3D0)
**To:** Everyone
**Re:** [16741488.519](https://www.hattrick.org/goto.ashx?path=%2FForum%2FRead.aspx%3Ft%3D16741488%26n%3D519%26v%3D0)
**Datetime:** 2015-09-08 22:03
**Message:**

> LA-MJ rašė:
>
> I've been thinking about absolute contributions today. To make better sense of this disaster: [(https://docs.google.com/spreadsheets/d/1bNwtB...gid=0)](https://docs.google.com/spreadsheets/d/1bNwtBdOxbY8pdY7uAx0boqHwRtJgj7tNpcFtDsP9Wq8/edit#gid=0) we need two ratios:
> 1) WBd defending: CD defending
> 2) Wo winger: FW scoring

The reason I started this is to improve the algorithm of Foxtrick's PlayerPositionsEvaluations module. The purpose of it is to accumulate a value that informs how good a certain player is at a certain position.
Applications:
1) find the best player for a certain position
2) find the best position for a certain player

Naturally to accumulate that value you sum up each relevant skill multiplied by a certain contribution factor. Currently it's a simple sum of contributions to all sectors. Obviously this is too simplified.

---

I exported the new logic and factors here:
[(https://docs.google.com/spreadsheets/d/11yWA_...aring)](https://docs.google.com/spreadsheets/d/11yWA_JiL_jwtb_SR28YlLHPJgv1DMHMVdMWHmFUWuuE/edit?usp=sharing)

Basically, wing attacks/defence are multiplied by WO:FW and WBD:CD ratios, respectively. This is to combat the uselessness of relative values given by HTs.

Then attack/defence ratings are averaged out across 3 sectors (25:35:25)

The defence is multiplied by 4/3 since the probability to score is ~3/7 when attack=defence.

Playmaking is also multiplied by 4/3. This is sort of empirical to get the proper IM player ordering for my team. It makes MF ~= 4 attack sectors importance-wise which is kinda logical.

An interesting consequence of trying to normalize attack across all 3 sectors is that winger skill is underestimated (it contributes to one side only). Midfield and/or defence seem to be more useful for wingers, lol.

Not sure what to do about it, if anything.
